### PR TITLE
ModemManager protocol cleanup and fixes on device removal handling

### PR DESCRIPTION
**Pull request by aleksander0m**
Maintainer: @nickberry17 

Description: Several fixes in the modemmanager protocol handler, including one that prevented from a proper reconnection if e.g. the device was unplugged and replugged, or if it reseted itself.

### CODE REVIEW

**Comment by jefferyto** [comment]
I suggest doing a force-push to trigger another CI check.

**Comment by amollad** [comment]
What is stopping this PR to get merged?

**Comment by feckert** [comment]
> What is stopping this PR to get merged?

Maybe no one has time to test this?

By the way.
I think the scripts that is sourced in /etc/hotplug.d/usb (modemmanager.common) should go to `/usr/share/modemmager` This is the right place. That is no configuration and so should not be placed under `/etc`

File: net/modemmanager/files/25-modemmanager-usb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# Copyright (C) 2019 Aleksander Morgado <aleksander@aleksander.es>
+
+# We need to process only full USB device removal events, we don't
+# want to process specific interface removal events.
+[ "$ACTION" = remove ] || exit
+[ -z "${INTERFACE}" ] || exit
+
+# Load common utilities
+. /etc/modemmanager/modemmanager.common
```
Comment (jow-) [inline net/modemmanager/files/25-modemmanager-usb]: Move this to `/usr/share` - it is code, not configuration

**Comment by jefferyto** [comment]
Apologies for the failing CI test - if you rebase on current master, it should be able to pass the download SDK step.

File: net/modemmanager/files/25-modemmanager-usb (hunk 1 above)
Comment (aleksander0m) [reply net/modemmanager/files/25-modemmanager-usb]: Done in a follow up commit.

**Comment by feckert** [comment]
From my side :+1: 

thanks for you contribution

**Pull request merged**